add-interactive.c

```diff
@@ -69,7 +71,14 @@ void init_add_i_state(struct add_i_state *s, struct repository *r)
 	git_config_get_string("diff.algorithm",
 			      &s->interactive_diff_algorithm);
 
-	git_config_get_bool("interactive.singlekey", &s->use_single_key);
+	git_config_get_bool("interactive.singleKey", &single_key);
+
+	if (single_key && !terminal_support(CBREAK)) {
+		warning(_("missing single keystrock support, "
+			  "disabling interactive.singleKey"));
```

Comment:
Translating the warning message is an improvement but I'd be tempted to use the original wording that is deleted below. It explains the problem in non-technical terms and avoids the issue that `interactive.singleKey` should not be translated within the message.